Packs/SMB/Integrations/SMB_v2/SMB_v2.py

```diff
@@ -182,6 +197,295 @@ def smb_rmdir(client: SMBClient, args: dict):
     return f"Directory: {path} was removed successfully"
 
 
+def smb_path_rename(client: SMBClient, args: dict):
+    """
+    Renames a file or directory on an SMB server.
+
+    Args:
+        client (SMBClient): An instance of the SMBClient class.
+        args (dict): Dictionary containing 'hostname', 'username', 'password', 'src_path', 'dest_path', and optional 'overwrite'.
+
+    Returns:
+        CommandResults: Result object with context and readable output.
+    """
+    hostname = args.get("hostname")
+    username = args.get("username")
+    password = args.get("password")
+    overwrite = args.get("overwrite", "no").lower()
+
+    source_path = handle_path(args.get("src_path"))
+    destination_path = handle_path(args.get("dest_path"))
+
+    source_path = create_share_path(hostname or client.hostname, source_path)
+    destination_path = create_share_path(hostname or client.hostname, destination_path)
+
+    client.create_session(hostname, username, password)
+
+    # Check if destination exists
+    dest_exists = False
+    try:
+        smb_stat(destination_path)
+        dest_exists = True
+    except Exception:
+        pass
+
+    if dest_exists and overwrite != "yes":
+        return CommandResults(
+            readable_output=f"Skipped rename: '{destination_path}' already exists and overwrite=no.",
+            outputs={
+                "SMBRename": {
+                    "OriginalPath": source_path,
+                    "NewPath": destination_path,
+                    "Renamed": False,
+                    "Reason": "Destination already exists and overwrite is disabled"
+                }
+            },
+            outputs_prefix="SMBRename",
+            outputs_key_field="OriginalPath"
+        )
+
+    # Perform rename
+    rename(source_path, destination_path)
+
+    return CommandResults(
+        readable_output=f"Successfully renamed '{source_path}' to '{destination_path}'",
+        outputs={
+            "SMBRename": {
+                "OriginalPath": source_path,
+                "NewPath": destination_path,
+                "Renamed": True
+            }
+        },
+        outputs_prefix="SMBRename",
+        outputs_key_field="OriginalPath"
+    )
+
+
+def smb_rename_dir(client: SMBClient, args: dict):
+    """
+    Renames a directory on an SMB server.
+
+    Args:
+        client (SMBClient): An instance of the SMBClient class.
+        args (dict): Dictionary containing 'hostname', 'username', 'password', 'src_path', and 'dest_path'.
+
+    Returns:
+        str: Success message indicating the directory was renamed.
+    """
+
+    hostname = args.get("hostname")
+    username = args.get("username")
+    password = args.get("password")
+    source_path = handle_path(args.get("src_path"))  # Ensuring clean paths
+    destination_path = handle_path(args.get("dest_path"))
```

Comment:
same regarding th args names, it's renaming not moving from source to destination